adapters/celestia/src/verifier/mod.rs

```diff
@@ -102,33 +105,53 @@ pub struct RollupParams {
     pub namespace: NamespaceId,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Hash)]
+/// A validity condition expressing that a chain of DA layer blocks is contiguous and canonical
+pub struct ChainValidityCondition {
+    pub prev_hash: [u8; 32],
+    pub block_hash: [u8; 32],
+}
+
+impl ValidityCondition for ChainValidityCondition {
+    fn combine<H: SimpleHasher>(&self, rhs: Self) -> Result<Self, anyhow::Error> {
```

Comment:
Nit: Should treat this code more as a library and use `thiserror` and let the caller convert to `anyhow`? The error enums in `thiserror` have better structure for examination by the caller of the api (so it will help with tests).

Reply:
Switched to an associated `Error` type in the trait and `ThisError` in the implementation.